lib/ddtrace/contrib/patcher.rb

```diff
@@ -6,21 +6,38 @@ module Contrib
     module Patcher
       def self.included(base)
         base.send(:include, Datadog::Patcher)
-        base.send(:extend, InstanceMethods)
-        base.send(:include, InstanceMethods)
+
+        base.singleton_class.send(:prepend, CommonMethods)
+        base.send(:prepend, CommonMethods) if base.class == Class
```

Comment:
Any particular reason for handling the case where the Patcher is a class, given all our current Patchers are modules?
Would it be for future proofing?

Reply:
Yeah, technically our integration API doesn't require all patchers to be modules, just that they include `Datadog::Contrib::Patcher` and behave like one. This means users who have created custom integrations might have implemented the patchers as classes: it's more of a "avoiding possible breaking change" thing than future proofing.